pkg/reconciler/inmemorychannel/inmemorychannel.go

```diff
@@ -0,0 +1,298 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package inmemorychannel
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/knative/eventing/pkg/apis/messaging/v1alpha1"
+	messaginginformers "github.com/knative/eventing/pkg/client/informers/externalversions/messaging/v1alpha1"
+	listers "github.com/knative/eventing/pkg/client/listers/messaging/v1alpha1"
+	"github.com/knative/eventing/pkg/logging"
+	"github.com/knative/eventing/pkg/reconciler"
+	"github.com/knative/pkg/controller"
+	"go.uber.org/zap"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	appsv1informers "k8s.io/client-go/informers/apps/v1"
+	corev1informers "k8s.io/client-go/informers/core/v1"
+	appsv1listers "k8s.io/client-go/listers/apps/v1"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+const (
+	// ReconcilerName is the name of the reconciler
+	ReconcilerName = "InMemoryChannels"
+
+	// controllerAgentName is the string used by this controller to identify
+	// itself when creating events.
+	controllerAgentName = "in-memory-channel-controller"
+
+	finalizerName = controllerAgentName
+
+	// Name of the corev1.Events emitted from the reconciliation process.
+	channelReconciled         = "InMemoryChannelReconciled"
```

Comment:
Is it K8s style to prefix the event names with the type they associate with? Or can we rely on the fact that the event targets a specific resource?

Reply:
that's what we seem to do in the other ones (Trigger, etc.)

Reply:
If there is no guideline prescribed here, then omitting the names would be better isn't it? When the event is recorded it is associated with a runtime object. So, the reason could omit InMemoryChannel.
Example when I get these events using kubectl, this is what I get
  Type    Reason             Age   From                          Message
  ----    ------             ----  ----                          -------
  Normal  ChannelReconciled  12s   in-memory-channel-controller  Channel reconciled: "msgdumper-im"

Reply:
removed the prefix.